registry/storage/filewriter.go

```diff
@@ -22,6 +23,11 @@ type fileWriter struct {
 	err    error // terminal error, if set, reader is closed
 }
 
+type bufferedFileWriter struct {
+	*fileWriter
```

Comment:
We do the same thing here and it works fine:

```
type layerUploadController struct {
	layerStore *layerStore

	uuid      string
	startedAt time.Time

	fileWriter
}
```

Why would it not work under these conditions?